_posts/2022-01-19-bazel-5.0.md

```diff
@@ -0,0 +1,148 @@
+* Removed `--incompatible_objc_compile_info_migration`. ([#10854](https://github.com/bazelbuild/bazel/issues/10854))
+* Removed `--incompatible_objc_provider_remove_compile_info`. ([#11359](https://github.com/bazelbuild/bazel/issues/11359))
+* Made gcov optional in `cc_toolchain` tools.
+* Added a new flag, `--incompatible_enable_cc_test_feature` which switches from the use of build variables to the feature of the same name.
+
+## Configurability
+
+* Added `--incompatible_enforce_config_setting_visibility` to make `config_setting` honor the `visibility` attribute (defaulting to `//visibility:public`) ([#12932](https://github.com/bazelbuild/bazel/issues/12932))
+* The flag `--toolchain_resolution_debug` now takes a regex argument, which is used to check which toolchain types or targets should have debug info printed. You may use `.*` as an argument to keep the current behavior of debugging every toolchain type / target.
+
+## Coverage
+
+* All (instead of just C++) source files are now filtered for coverage output according to `--instrumentation_filter` and `--instrument_test_targets`.
+* `genrule.srcs` is now considered a source attribute for coverage.
+* Forward coverage-instrumented files from non-tool dependencies by default.
+* `label_keyed_string_dict` attributes are now considered when gathering instrumented files for coverage.
+* Java branch coverage now applies Jacoco’s coverage filters. ([#12696](https://github.com/bazelbuild/bazel/issues/12696))
+
+## Java
+
+* Removed `javac_jar` from JavaToolchainInfo.
+* Flipped `--incompatible_disallow_resource_jars`, disabling `resouce_jars` attribute on `java_library`. ([#13221](https://github.com/bazelbuild/bazel/issues/13221))
+* Flipped `--incompatible_java_common_parameters`, removing `host_javabase` parameter from `java_common.compile`. ([#12373](https://github.com/bazelbuild/bazel/issues/12373))
+* Flipped `--incompatible_use_toolchain_resolution_for_java_rules`. Flags `--javabase` and `--java_toolchain` are noop. ​​([#7849](https://github.com/bazelbuild/bazel/issues/7849))
+* Only native libraries supported by C++ targets in `deps` and `runtime_deps` are collected. No longer collecting `.so` files ([#13043](https://github.com/bazelbuild/bazel/issues/13043)) or libraries in the `data` attribute. ([#13550](https://github.com/bazelbuild/bazel/issues/13550))
+* `JavaToolchainInfo.jvm_opt` returns a `depset` instead of a list.
+* Added a `_direct_source_jars` output group to Java related targets.
+* Added an `enable_annotation_processing` option to `java_common.compile`, which can be used to disable any annotation processors passed to `plugins` or in `exported_plugins` of `deps`.
+* Removed `java_lite_proto_library.strict_deps` attribute.
+* The Starlark implementations of `java_library` and `java_plugin` are now used instead of their native counterparts.
```

Comment:
This  feels like it needs more words.
Does that just mean that bazel itself does not use them?
Does it mean that java_library is going away in the next release. 
cc: @comius

Reply:
It means that Bazel now uses the Starlark impls of these rules (not `rules_java`, but something shipped with Bazel), instead of native Java impls. I've switched the statement to use the active voice. (btw Ivo wrote these)